Failed to compile on 1a038865

```
$ cargo install --git 'https://github.com/aome510/spotify-player' --no-default-features --features pulseaudio-backend,lyric-finder,clipboard,media-control,notify
...
   Compiling config_parser2 v0.1.5
error[E0446]: private type `ServiceState` in public interface
  --> /home/meow/.local/share/cargo/registry/src/index.crates.io-6f17d22bba15001f/souvlaki-0.7.0/src/platform/mpris/dbus/interfaces.rs:19:1
   |
19 | / pub fn register_methods<F>(
20 | |     state: &Arc<Mutex<ServiceState>>,
21 | |     event_handler: &Arc<Mutex<F>>,
22 | |     friendly_name: String,
...  |
25 | | where
26 | |     F: Fn(MediaControlEvent) + Send + 'static,
   | |______________________________________________^ can't leak private type
   |
  ::: /home/meow/.local/share/cargo/registry/src/index.crates.io-6f17d22bba15001f/souvlaki-0.7.0/src/platform/mpris/dbus/mod.rs:54:1
   |
54 |   struct ServiceState {
   |   ------------------- `ServiceState` declared as private

For more information about this error, try `rustc --explain E0446`.
error: could not compile `souvlaki` (lib) due to previous error
warning: build failed, waiting for other jobs to finish...
error: failed to compile `spotify_player v0.16.3 (https://github.com/aome510/spotify-player#1a038865)`, intermediate artifacts can be found at `/tmp/cargo-installpP5ipQ`.
To reuse those artifacts with a future compilation, set the environment variable `CARGO_TARGET_DIR` to that path.
```

--- a/lyric_finder/Cargo.toml
+++ b/lyric_finder/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "lyric_finder"
-version = "0.1.4"
+version = "0.1.5"
 edition = "2021"
 license = "MIT"
 description = "A lyric finder library"
@@ -9,16 +9,16 @@ repository = "https://github.com/aome510/spotify-player/tree/master/lyric_finder
 keywords = ["lyric", "song", "track"]
 
 [dependencies]
-anyhow = "1.0.71"
-reqwest = { version = "0.11.18", features = ["json"] }
-serde = { version = "1.0.163", features = ["derive"] }
+anyhow = "1.0.79"
+reqwest = { version = "0.11.24", features = ["json"] }
+serde = { version = "1.0.196", features = ["derive"] }
 html5ever = "0.26.0"
 markup5ever_rcdom = "0.2.0"
-log = "0.4.18"
+log = "0.4.20"
 
 [dev-dependencies]
-tokio = { version = "1.28.2", features = ["rt", "rt-multi-thread", "macros"] }
-env_logger = "0.10.0"
+tokio = { version = "1.36.0", features = ["rt", "rt-multi-thread", "macros"] }
+env_logger = { version = "0.11.1", default-features = false }
 
 [[example]]
 name = "lyric-finder"

--- a/spotify_player/Cargo.toml
+++ b/spotify_player/Cargo.toml
@@ -10,8 +10,8 @@ keywords = ["spotify", "tui", "player"]
 readme = "../README.md"
 
 [dependencies]
-anyhow = "1.0.75"
-clap = { version = "4.4.11", features = ["derive", "string"] }
+anyhow = "1.0.79"
+clap = { version = "4.4.18", features = ["derive", "string"] }
 config_parser2 = "0.1.5"
 crossterm = "0.27.0"
 dirs-next = "2.0.0"
@@ -19,38 +19,38 @@ librespot-connect = { version = "0.4.2", optional = true }
 librespot-playback = { version = "0.4.2", optional = true }
 librespot-core = "0.4.2"
 log = "0.4.20"
-chrono = "0.4.31"
-reqwest = { version = "0.11.22", features = ["json"] }
+chrono = "0.4.33"
+reqwest = { version = "0.11.24", features = ["json"] }
 rpassword = "7.3.1"
 rspotify = "0.12.0"
-serde = { version = "1.0.193", features = ["derive"] }
-tokio = { version = "1.35.0", features = ["rt", "rt-multi-thread", "macros", "time"] }
-toml = "0.8.8"
-tui = { package = "ratatui", version = "0.25.0" }
+serde = { version = "1.0.196", features = ["derive"] }
+tokio = { version = "1.36.0", features = ["rt", "rt-multi-thread", "macros", "time"] }
+toml = "0.8.9"
+tui = { package = "ratatui", version = "0.26.0" }
 rand = "0.8.5"
-maybe-async = "0.2.7"
-async-trait = "0.1.74"
+maybe-async = "0.2.9"
+async-trait = "0.1.77"
 parking_lot = "0.12.1"
 tracing = "0.1.40"
 tracing-subscriber = { version = "0.3.18", features = ["env-filter"] }
-lyric_finder = { version = "0.1.4", path = "../lyric_finder" , optional = true }
+lyric_finder = { version = "0.1.5", path = "../lyric_finder" , optional = true }
 backtrace = "0.3.69"
-souvlaki = { version = "0.7.0", optional = true }
-winit = { version = "0.29.4", optional = true }
+souvlaki = { version = "0.7.3", optional = true }
+winit = { version = "0.29.10", optional = true }
 viuer = { version = "0.7.1", optional = true }
-image = { version = "0.24.7", optional = true }
+image = { version = "0.24.8", optional = true }
 notify-rust = { version = "4.10.0", optional = true, default-features = false, features = ["d"] }
 flume = "0.11.0"
-serde_json = "1.0.108"
+serde_json = "1.0.113"
 once_cell = "1.19.0"
-regex = "1.10.2"
+regex = "1.10.3"
 daemonize = { version = "0.5.0", optional = true }
 ttl_cache = "0.5.1"
 copypasta = { version = "0.10.0", optional = true }
-clap_complete = "4.4.4"
+clap_complete = "4.4.10"
 
 [target.'cfg(target_os = "windows")'.dependencies.windows]
-version = "0.44.0"
+version = "0.52.0"
 features = [
 	"Win32_Foundation",
 	"Win32_Graphics_Gdi",

--- a/spotify_player/src/media_control.rs
+++ b/spotify_player/src/media_control.rs
@@ -144,8 +144,8 @@ mod windows {
     use std::io::Error;
     use std::mem;
 
+    use windows::core::w;
     use windows::core::PCWSTR;
-    use windows::w;
     use windows::Win32::Foundation::{HWND, LPARAM, LRESULT, WPARAM};
     use windows::Win32::System::LibraryLoader::GetModuleHandleW;
     use windows::Win32::UI::WindowsAndMessaging::{
@@ -168,7 +168,7 @@ mod windows {
 
                 let wnd_class = WNDCLASSEXW {
                     cbSize: mem::size_of::<WNDCLASSEXW>() as u32,
-                    hInstance: instance,
+                    hInstance: instance.into(),
                     lpszClassName: class_name,
                     lpfnWndProc: Some(Self::wnd_proc),
                     ..Default::default()
@@ -221,7 +221,7 @@ mod windows {
     impl Drop for DummyWindow {
         fn drop(&mut self) {
             unsafe {
-                DestroyWindow(self.handle);
+                DestroyWindow(self.handle).unwrap();
             }
         }
     }

--- a/spotify_player/src/ui/playback.rs
+++ b/spotify_player/src/ui/playback.rs
@@ -152,7 +152,7 @@ fn render_playback_text(
     // based on a user-configurable format string (app_config.playback_format)
     let format_str = &state.configs.app_config.playback_format;
 
-    let mut playback_text = Text { lines: vec![] };
+    let mut playback_text = Text::default();
     let mut spans = vec![];
 
     // this regex is to handle a format argument or a newline

